src/test/scala/org/ergoplatform/network/ErgoNodeViewSynchronizerSpecification.scala

```diff
@@ -226,16 +247,32 @@ class ErgoNodeViewSynchronizerSpecification extends HistoryTestHelpers with Matc
   }
 
   property("NodeViewSynchronizer: receiving out-of-order header should request it again") {
-    withFixture { ctx =>
+    withFixture2 { ctx =>
       import ctx._
+
+      implicit val patienceConfig: PatienceConfig = PatienceConfig(5.seconds, 100.millis)
+
+      def sendHeader(synchronizerRef: ActorRef,  header: Header): Unit = {
+        deliveryTracker.setRequested(Seq(header.id), Header.modifierTypeId, Some(peer))(_ => Cancellable.alreadyCancelled)
+        val modData = ModifiersData(Header.modifierTypeId, Map(header.id -> header.bytes))
+        val modSpec = new ModifiersSpec(100)
+        synchronizerRef ! Message(modSpec, Left(modSpec.toBytes(modData)), Some(peer))
+      }
+
       deliveryTracker.reset()
-      deliveryTracker.setRequested(Seq(chain.last.id), Header.modifierTypeId, Some(peer))(_ => Cancellable.alreadyCancelled)
-      val modData = ModifiersData(Header.modifierTypeId, Map(chain.last.id -> chain.last.bytes))
-      val modSpec = new ModifiersSpec(100)
-      node ! Message(modSpec, Left(modSpec.toBytes(modData)), Some(peer))
-      // desired state of submitting headers out of order is Unknown, they need to be downloaded again
+
+      // we generate fork of two headers, starting from the parent of the best header
+      // so the depth of the rollback is 1, and the fork bypasses the best chain by 1 header
```

Comment:
Why do we need a fork for reproducing this?

Reply:
We need for a header @ height of best chain height + 1, otherwise the header will be applied to cache just and not validated inside `pmodModify` (where `RecoverableModifierError` is thrown). To test out-ot-order application then, we need to generate and a fork which first forked header is @ best chain height, and then there's the header we are trying to apply (@ height of best chain + 1) without presenting its parent first.